app/assets/javascripts/modules/inline-image-modal.js

```diff
@@ -109,20 +116,37 @@ InlineImageModal.prototype.performAction = function (item) {
         }.bind(this))
     },
     'meta': function () {
-      this.postModalForm(item.dataset.modalActionForm)
-        .then(function (text) {
-          this.$multiSectionViewer.showDynamicSection(text)
+      this.postModalForm(item)
+        .then(function (result) {
+          this.$multiSectionViewer.showDynamicSection(result.body)
           this.overrideActions()
           this.initComponents()
         }.bind(this))
         .catch(function (result) {
           this.$multiSectionViewer.showStaticSection('error')
         }.bind(this))
     },
+    'metaInsert': function () {
+      this.postModalForm(item)
+        .then(function (result) {
+          if (result.finished) {
+            var editor = this.$module.closest('[data-module="markdown-editor"]')
+            this.$modal.close()
+            editor.selectionReplace(item.dataset.modalData)
+          } else {
+            this.$multiSectionViewer.showDynamicSection(result.body)
+            this.overrideActions()
+            this.initComponents()
```

Comment:
I wonder if you're getting to a point where it'd be useful to have a single function that does the show, override and init?